### DISCUSSION TITLE
[v7] and MobX - looking for an advice

### DISCUSSION THREAD

**Discussion by MaximBG** [TanStack/table#2437] [General]
In my application react-table displays data from a MobX store. The records in the store change over time. Sometimes I replace the whole record and sometimes there is just some property changes so I update just it.
Recently I came to a problem which is described here: https://github.com/tannerlinsley/react-table/issues/2397
I have couple of questions with regard to it. The link to the codesandbox is inside the issue, but please note, that by default it shows the failure, and in order to eliminate error you should either comment out useSortBy hook, or downgrade react-table to 7.0.0-rc.1

I found, that I can overcome my issue if I dereference the data earlier, but then I lose the ability to update values inside the records. (There are two tables side-by-side that show both approaches) So while the MobX recommends to ["dereference values as late as possible"](https://mobx.js.org/best/pitfalls.html) maybe react-table has a problem to deal with mutable data and I should use the workaround above?

I thought that I can use 7.0.0-rc.1, but then I found that even there the "Maximum update depth exceeded" error pops up if I try to resize a column while the useSortBy hook present.

What are the chances, that this issue will be fixed?  A similar [issue](https://github.com/tannerlinsley/react-table/issues/2369) was opened two weeks ago and recently there was a release that didn't include the fix.

**Comment by thomaszdxsn** [comment]
same issue +1

**Comment by thomaszdxsn** [comment]
I found a workaround, i try to `toJS` in the same block for my observable, like:

```
  get PBSTableData() {
    console.log(toJS(this.activityRoot))
    return [this.activityRoot]
  }
```

so, all observable is read.. it is don't a best practice but work
